src/main/java/com/chargerrobotics/utils/ArduinoSerial.java

```diff
@@ -6,30 +6,43 @@
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 
+import com.chargerrobotics.utils.ArduinoSerialReceiver.ArduinoListener;
 import com.fazecast.jSerialComm.SerialPort;
+import com.fazecast.jSerialComm.SerialPortIOException;
+import com.fazecast.jSerialComm.SerialPortTimeoutException;
 
-public abstract class ArduinoSerial {
+public class ArduinoSerial {
 
 	private final SerialPort serial;
 	private boolean isOpen;
 	private InputStream in;
 	private OutputStream out;
 
-	private static final long TIMEOUT = 2000;
-	
-	private static final byte[] sync = new byte[] { 85, 85 };
-	private static final byte[] poll = new byte[] { -127, 66 };
-	private static final byte[] response = new byte[] { 45, -120 };
+	private static final long TIMEOUT = 5;
+	private static final long RESPONSE_TIMEOUT = 20;
+
+	private static final byte[] sync = new byte[] { 0x55, 0x55 };
+	private static final byte[] poll = new byte[] { (byte) 0x81, 0x42 };
 	
-	private static final Checksum cs = new Checksum();
-	private final ByteBuffer dataBuffer = ByteBuffer.allocate(256).order(ByteOrder.LITTLE_ENDIAN);
+	private final String name;
+
+	private static final Checksum recCs = new Checksum();
+	private static final Checksum sendCs = new Checksum();
+	private final ByteBuffer recBuffer = ByteBuffer.allocate(256).order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
I found the information on thread safety, and it sounds like bytebuffers used by multiple threads need to have all accesses properly synchronized. Here's what the documentation for the Buffer class says:

`Buffers are not safe for use by multiple concurrent threads. If a buffer is to be used by more than one thread then access to the buffer should be controlled by appropriate synchronization.`

Reply:
Talked about it in person, and the synchronization needed is on the outward side of the listeners, not the serial-facing side. This should be fine, we just need to be careful in our listeners